code/modules/cooking/machinery/cooking_machines/_cooker.dm

```diff
@@ -78,19 +128,19 @@
 	var/temp_scale = 0
 	if(temperature > min_temp)
 		if(temperature >= optimal_temp)
-			temp_scale = 1
+			temp_scale = Clamp(1 - ((optimal_temp - temperature) / optimal_temp), 0, 1)
 		else
-			temp_scale = (temperature - 73.15) / (optimal_temp - 73.15)
+			temp_scale = temperature / optimal_temp
 		//If we're between min and optimal this will yield a value in the range 0.7 to 1
 
 	cooking_coeff = optimal_power * temp_scale
-	RefreshParts()
+	RefreshParts() // this is what actually updates the cooking power, for some reason.
 
 /obj/machinery/appliance/cooker/proc/heat_up()
-	if (temperature < optimal_temp)
-		if (use_power == 1 && ((optimal_temp - temperature) > 5))
+	if (temperature < set_temp)
+		if (use_power == 1 && ((set_temp - temperature) > 5))
```

Comment:
Magic numbers, use_power should have defines

Reply:
I thought those were Bay-only, I recall checking and not finding them.